lib/vrl/stdlib/src/chunks.rs

```diff
@@ -0,0 +1,134 @@
+                required: true,
+            },
+            Parameter {
+                keyword: "chunk_size",
+                kind: kind::INTEGER,
+                required: true,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "chunks by byte",
+                source: r#"chunks("abcdefgh", 4)"#,
+                result: Ok(r#"["abcd", "efgh"]"#),
+            },
+            Example {
+                title: "chunk size must be at least 1 byte",
+                source: r#"chunks("hello world", 0)"#,
+                result: Err(
+                    r#"function call error for "chunks" at (0:24): "chunk_size" must be at least 1 byte"#,
+                ),
+            },
+            Example {
+                title: "chunk sizes do not respect unicode code point boundaries",
+                source: r#"chunks("ab你好", 4)"#,
+                result: Ok(r#"["ab�","�好"]"#),
+            },
+        ]
+    }
+
+    fn compile(
+        &self,
+        _state: (&mut state::LocalEnv, &mut state::ExternalEnv),
+        _ctx: &mut FunctionCompileContext,
+        mut arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let chunk_size = arguments.required("chunk_size");
+
+        Ok(Box::new(ChunksFn { value, chunk_size }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ChunksFn {
+    value: Box<dyn Expression>,
+    chunk_size: Box<dyn Expression>,
+}
+
+impl Expression for ChunksFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+        let chunk_size = self.chunk_size.resolve(ctx)?;
+
+        chunks(value, chunk_size)
+    }
+
+    fn type_def(&self, _state: (&state::LocalEnv, &state::ExternalEnv)) -> TypeDef {
+        let chunk_size_is_expression = self.chunk_size.as_value().is_none();
+        // chunk_size is converted to a usize later, so if a user-supplied Value::Integer (i64) is
+        // larger than the platform's usize::MAX, it could fail to convert.
+        let chunk_size_limited_by_architecture = usize::BITS < 64;
+
+        TypeDef::array(Collection::from_unknown(Kind::bytes()))
+            .with_fallibility(chunk_size_is_expression || chunk_size_limited_by_architecture)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    test_function![
+        chunks => Chunks;
+
+        chunks_data {
+            args: func_args![value: "abcdefgh",
+                             chunk_size: 4,
+            ],
+            want: Ok(value!(["abcd", "efgh"])),
+            tdef: TypeDef::array(Collection::from_unknown(Kind::bytes())),
+        }
+
+        minimum_chunk_size {
+            args: func_args![value: "",
+                             chunk_size: 0,
+            ],
+            want: Err(r#""chunk_size" must be at least 1 byte"#),
+            tdef: TypeDef::array(Collection::from_unknown(Kind::bytes())),
```

Comment:
This is an invalid type definition. (The function had an error, but the type definition is infallible)

Reply:
This seems to pass either way in the latest commit. Since it's now a compile time error in ce0e9b6ee6d0ed5000108b1cb31137a56568e8ed, should it be marked as fallible?

I considered taking this test out as well since I covered it in `chunks_chunk_size_at_least_1.vrl`, since it's a compile time error.

Reply:
I'd just remove it from here since it's covered in the VRL test already. I don't think these tests handle compile time errors well yet.